src/impl/regexParser.js

```diff
@@ -189,6 +190,34 @@ function extractASCII(match) {
   return [result, FixedOffsetZone.utcInstance];
 }
 
+const sqlDateRegex = /^(\d{4})-(\d\d)-(\d\d)$/,
+  sqlTimeRegex = /^(\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/,
+  sqlDateTimeRegex = /^(\d{4})-(\d\d)-(\d\d) (\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/;
+
+function extractSqlDate(match) {
+  const [, yearStr, monthStr, dayStr] = match,
+    result = fromStrings(0, yearStr, monthStr, dayStr, 0, 0, 0);
+  return [result, FixedOffsetZone.utcInstance];
+}
+
+function extractSqlTime(match) {
+  const now = new Date(),
+    yearStr = now.getUTCFullYear().toString(),
+    monthStr = (now.getUTCMonth() + 1).toString(),
+    dayStr = now.getUTCDate().toString(),
+    [, hourStr, minuteStr, secondStr] = match,
+    result = fromStrings(0, yearStr, monthStr, dayStr, hourStr, minuteStr, secondStr);
+
+  return [result, FixedOffsetZone.utcInstance];
+}
+
+function extractSqlDateTime(match) {
```

Comment:
You're not having these things take a cursor argument. But if you did have `extractSQLTime` and `extractSQLDate` use `slice` with a cursor argument, you could actually use `combineExtractors` instead of this function. If that's too unclear, don't worry about it; it's my fault for making this code too difficult to understand and this seems fine.

Reply:
yeah i want to use that too, but from what i understand,  it will combine without the empty space i need in the datetime regex. I'm not sure if my understanding is correct here hence i did not use it.

`const full = regexes.reduce((f, r) => f + r.source, '');`

Reply:
Yeah, you need to use a separate regex because of the space. But you don't need to use a separate extractor because you don't capture the space in a capture group. Extractors only care about what units correspond to what positions in the match array.

Now that I think about it, that actually makes me wonder if you can actually just reuse the ISO extractors as-is? You need separate regexes, but aren't the units in the same order?